conans/test/functional/toolchains/cmake/test_universal_binaries.py

```diff
@@ -0,0 +1,83 @@
+import platform
+import textwrap
+
+import pytest
+
+from conans.test.utils.tools import TestClient
+
+
+@pytest.mark.skipif(platform.system() != "Darwin", reason="Only OSX")
+#@pytest.mark.tool("cmake")
+def test_create_universal_binary():
+    client = TestClient()
+
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.cmake import CMake, cmake_layout
+        class mylibraryRecipe(ConanFile):
+            package_type = "library"
+            generators = "CMakeToolchain"
+            settings = "os", "compiler", "build_type", "arch"
+            options = {"shared": [True, False], "fPIC": [True, False]}
+            default_options = {"shared": False, "fPIC": True}
+            exports_sources = "CMakeLists.txt", "src/*", "include/*"
+
+            def layout(self):
+                cmake_layout(self)
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+                cmake.build()
+                self.run("lipo -info libmylibrary.a")
+
+            def package(self):
+                cmake = CMake(self)
+                cmake.install()
+
+            def package_info(self):
+                self.cpp_info.libs = ["mylibrary"]
+    """)
+
+    test_conanfile = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+        from conan.tools.cmake import CMake, cmake_layout
+        from conan.tools.build import can_run
+
+        class mylibraryTestConan(ConanFile):
+            settings = "os", "compiler", "build_type", "arch"
+            generators = "CMakeDeps", "CMakeToolchain"
+
+            def requirements(self):
+                self.requires(self.tested_reference_str)
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+                cmake.build()
+
+            def layout(self):
+                cmake_layout(self)
+
+            def test(self):
+                # leaving can_run here to test the real case
+                if can_run(self):
+                    exe = os.path.join(self.cpp.build.bindir, "example")
+                    self.run(exe, env="conanrun")
+                    self.run(f"lipo {exe} -info", env="conanrun")
+            """)
+
+    client.run("new cmake_lib -d name=mylibrary -d version=1.0")
+    client.save({"conanfile.py": conanfile, "test_package/conanfile.py": test_conanfile})
+
+    client.run('create . --name=mylibrary --version=1.0 '
+               '-s="arch=x86_64-armv8-armv8.3" --build=missing -tf=""')
```

Comment:
I used "-" as a separator because it's not problematic when the paths are composed in the layouts. 
Only elbrus archs use the "-" and those can't be used for universal binaries.

Reply:
Maybe we should add something so that `x86_64-armv8-armv8.3` and `armv8-x86_64-armv8.3` are not different? or order alphabetically and throw an error?

Reply:
would `;` work? 
when composing paths in layouts we can always "replace" the separator with a character that is valid in paths? - it does read fairly strange

Reply:
Yes, that would work, I tried with `;` before because it reads better than `-` and closer to the thing you put in CMake but I reverted because of the paths, but we can change it to `;` again and replace in the end when composing the paths.

Reply:
finally, looks like the `|` is not problematic and looks better than the `-`: `armv8|armv8.3|x86_64`